gitlabform/gitlabform/core.py

```diff
@@ -102,26 +124,40 @@ def parse_args(self):
 
         parser.add_argument('-n', '--noop', dest='noop', action="store_true", help='Run in no-op (dry run) mode')
 
-        return parser.parse_args()
+        args = parser.parse_args()
 
-    def set_log_level(self):
+        return args.project_or_group, args.config, args.verbose, args.debug, args.strict, args.start_from, args.noop
+
+    def set_log_level(self, tests=False):
 
         logging.basicConfig()
         level = logging.WARNING
-        if self.args.verbose:
+        if self.verbose:
             level = logging.INFO
-        elif self.args.debug:
+        elif self.debug:
             level = logging.DEBUG
         logging.getLogger().setLevel(level)
 
-        fmt = logging.Formatter("%(message)s")
-        logging.getLogger().handlers[0].setFormatter(fmt)
+        if not tests:
+            fmt = logging.Formatter("%(message)s")
+            logging.getLogger().handlers[0].setFormatter(fmt)
+        else:
+            # disable printing to stdout/err because pytest will catch it anyway
+            handler = logging.getLogger().handlers[0]
+            logging.getLogger().removeHandler(handler)
 
     def initialize_configuration_and_gitlab(self):
 
         try:
-            gl = GitLab(self.args.config.strip())
-            c = Configuration(self.args.config.strip())
+            if hasattr(self, 'config_string'):
+                gl = GitLab(config_string=self.config_string)
```

Comment:
why don't you put both `gl = GitLab(config_string=self.config_string)` and `c = Configuration(config_string=self.config_string)` in this `if`?

Reply:
Done, thanks!